crates/wysiwyg/src/dom/nodes/container_node.rs

```diff
@@ -374,6 +381,265 @@ where
     }
 }
 
+#[cfg(feature = "to-markdown")]
+impl<S> ToMarkdown<S> for ContainerNode<S>
+where
+    S: UnicodeString,
+{
+    fn fmt_markdown(
+        &self,
+        buffer: &mut S,
+        options: &MarkdownOptions,
+    ) -> Result<(), MarkdownError<S>> {
+        use ContainerNodeKind::*;
+        use InlineFormatType::*;
+
+        // `fmt_children` is a super basic loop over children to call
+        // `fmt_markdown`, except that it inserts `\n` between block
+        // nodes.
+        fn fmt_children<S>(
+            this: &ContainerNode<S>,
+            buffer: &mut S,
+            options: &MarkdownOptions,
+        ) -> Result<(), MarkdownError<S>>
+        where
+            S: UnicodeString,
+        {
+            for (nth, child) in this.children.iter().enumerate() {
+                if nth > 0 && child.is_block_node() {
+                    buffer.push("\n");
+                }
+
+                child.fmt_markdown(buffer, options)?;
+            }
+
+            Ok(())
+        }
+
+        let mut options = *options;
+
+        match self.kind() {
```

Comment:
This got long. What do you think about breaking the match arms into separate functions?

Reply:
Functions will be super small, and the `match` will stay. What's long is the comments, not the code. If you want me to change, that's fine, but I'm not sure it gonna make things any simpler: more code (due to function declaration), and more jump in the code.

Reply:
Yeah, I do want it broken up please. Long functions, even if it's comments making them long, make it hard to follow the flow.

Reply:
Done, https://github.com/matrix-org/matrix-wysiwyg/pull/143/commits/568250f784fa180fbc0343b3e40e821349fce322.